server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1930,6 +1960,22 @@ public void onFailure(Exception e) {
         return true;
     }
 
+    /**
+     * Tries to find the latest cluster UUID that wrote to this repository on a best effort basis by listing out repository root contents
+     * to find the latest repository generation and then reading the cluster UUID of the last writer from the {@link RepositoryData} found
+     * at this generation.
+     *
+     * @return cluster UUID of the last cluster to write to this repository or {@code null} if none was recorded
+     */
+    @Nullable
+    private Tuple<Long, String> readLastWriterInfo() throws IOException {
+        assert bestEffortConsistency == false : "This should only be used for adding information to errors in consistent mode";
+        final long latestGeneration = latestIndexBlobId();
+        final RepositoryData actualRepositoryData = getRepositoryData(latestGeneration);
+        return Tuple.tuple(latestGeneration,
+                actualRepositoryData.getClusterUUID().equals(MISSING_UUID) == false ? actualRepositoryData.getClusterUUID() : null);
```

Comment:
Maybe just use `MISSING_UUID` to indicate it's missing everywhere?

Reply:
++